File: pathfinder-api/src/main/java/de/cubbossa/pathapi/storage/cache/DiscoverInfoCache.java
package de.cubbossa.pathapi.storage.cache;

import de.cubbossa.pathapi.misc.NamespacedKey;
import de.cubbossa.pathapi.storage.DiscoverInfo;

import java.util.Collection;
import java.util.Optional;
import java.util.UUID;

public interface DiscoverInfoCache extends StorageCache<DiscoverInfo> {

  Optional<DiscoverInfo> getDiscovery(UUID player, NamespacedKey key);

  Optional<Collection<DiscoverInfo>> getDiscovery(UUID player);

  void invalidate(UUID player);
}

File: pathfinder-api/src/main/java/de/cubbossa/pathfinder/storage/cache/DiscoverInfoCache.kt
package de.cubbossa.pathfinder.storage.cache

import de.cubbossa.pathfinder.misc.NamespacedKey
import de.cubbossa.pathfinder.storage.DiscoverInfo
import java.util.*

interface DiscoverInfoCache : StorageCache<DiscoverInfo> {
    fun getDiscovery(player: UUID, key: NamespacedKey): DiscoverInfo?

    fun getDiscovery(player: UUID): Collection<DiscoverInfo>?

    fun invalidate(player: UUID)
}
